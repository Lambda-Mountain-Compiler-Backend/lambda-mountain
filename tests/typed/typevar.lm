
f := Î»(: x tx)(: y ty). x;

f( 1 2 )
