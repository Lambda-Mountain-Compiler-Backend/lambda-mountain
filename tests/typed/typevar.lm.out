((() (Global (f (: ((App ((App ((App ((App ((Literal :) (Variable x))) (Variable tx))) (App ((App ((Literal :) (Variable y))) (Variable ty))))) (: ((Variable x) tx)))) (-> ((Cons (tx ty)) tx))))))) (GExpr (: ((App ((: ((Variable f) (-> ((Cons (tx ty)) tx)))) (: ((App ((: ((Literal 1) Atom)) (: ((Literal 2) Atom)))) (Cons (Atom Atom)))))) Atom))))
